docs/high/file.rst

```diff
@@ -465,10 +465,19 @@ Reference
     :param min_raw_keep: Minimum percentage of raw data to keep in the page
             buffer before allowing pages containing raw data to be evicted.
             Applicable only if ``page_buf_size`` is set. Default value is zero.
-    :param locking: The file locking behavior. One of False (or "false"), True
-            (or "true"), "best-effort", or None. Warning: The
-            HDF5_USE_FILE_LOCKING environment variable can override this
-            parameter. Only available with HDF5 >= 1.12.1 or 1.10.x >= 1.10.7.
+    :param locking: The file locking behavior. One of:
+
+            - False (or "false") --  Disable file locking
+            - True (or "true")   --  Enable file locking
+            - "best-effort"      --  Enable file locking but ignore some errors
+            - None               --  Use HDF5 defaults
```

Comment:
Is it worth saying what the default is (locking enabled, as far as I know)? I know it could change, but it's a minor annoyance when docs tell me that the default is the default and I have to look elsewhere to find what it actually is.

Reply:
Default is currently "best-effort" and should remain so until I rip the scheme out.

Reply:
Thanks! Out of interest, what do you mean about ripping the scheme out? Is that the best-effort scheme specifically, or the whole locking scheme? If the latter, will it be replaced with anything?

Reply:
I'd like to rip out the entire file locking scheme since it seems to cause more problems than it solves. There would be no replacement scheme, so users would just have to be careful about accessing files.

Keep in mind that the configure option is from 1.10.7 and 1.12.1 and later. There is a matrix here:

https://github.com/HDFGroup/hdf5/blob/develop/doc/file-locking.md

Reply:
Thanks! We mention the version restriction just below this list.

I tried recently, and it seems that IBM Spectrum Scale (aka GPFS) doesn't share flock locks across nodes, whereas lockf/fcntl locks are shared. So it's already possible to have the same HDF5 file open for writing on multiple nodes simultaneously, even where the lock calls work.

Maybe there's an option that a sysadmin can set somewhere in GPFS to adjust this behaviour; I'm just going on what happens in one of our GPFS filesystems.

Reply:
so the task are :
 - add the default
 - note it may be removed in the future
 - add the link  https://github.com/HDFGroup/hdf5/blob/develop/doc/file-locking.md

Reply:
Also, FWIW, the .md file will eventually be incorporated in the Doxygen-based HDF5 user guide (included in the repo). I'll probably keep the developer-level bits around in the MD file, though.